lib/booking.rb

```diff
@@ -0,0 +1,43 @@
+class Booking
+  attr_reader :start_date, :end_date, :confirmed, :userid, :spaceid
+
+  def initialize(start_date:, end_date:, userid:, spaceid:, confirmed:)
+    @start_date = start_date
+    @end_date = end_date
+    @userid = userid
+    @spaceid = spaceid
+    @confirmed = confirmed
+  end
+
+  def self.create_booking(userid:, spaceid:, start_date:, end_date: )
+    result = DatabaseConnection.query("INSERT
+      INTO bookings (spaceid, hirerid, startdate, enddate)
+      VALUES (#{spaceid}, #{userid}, '#{start_date}', '#{end_date}')
+      RETURNING spaceid, hirerid, startdate, enddate, confirmed;
+      ")
+
+      Booking.new(
+        start_date: result[0]['startdate'],
+        end_date: result[0]['enddate'],
+        userid: result[0]['hirerid'],
+        spaceid: result[0]['spaceid'],
+        confirmed: false
```

Comment:
This line is substituting the data returned from the table - it should be false by default but that aspect of the test is redundant if the falseness is declared at this point.
```suggestion
        confirmed: result[0]['confirmed']
```